besu/src/main/java/org/hyperledger/besu/controller/ConsensusScheduleBesuControllerBuilder.java

```diff
@@ -105,7 +115,46 @@ protected MiningCoordinator createMiningCoordinator(
 
   @Override
   protected ProtocolSchedule createProtocolSchedule() {
-    return besuControllerBuilderSchedule.get(0L).createProtocolSchedule();
+    final NavigableSet<BftForkSpec<ProtocolSchedule>> forkSpecs =
+        besuControllerBuilderSchedule.entrySet().stream()
+            .map(
+                entry ->
+                    new BftForkSpec<>(entry.getKey(), entry.getValue().createProtocolSchedule()))
+            .collect(
+                Collectors.toCollection(
+                    () -> new TreeSet<>(Comparator.comparing(BftForkSpec::getBlock))));
+
+    final MutableProtocolSchedule combinedProtocolSchedule =
+        new MutableProtocolSchedule(genesisConfig.getConfigOptions().getChainId());
+    for (BftForkSpec<ProtocolSchedule> spec : forkSpecs) {
+      checkState(
+          spec.getConfigOptions() instanceof MutableProtocolSchedule,
+          "Consensus migration requires a MutableProtocolSchedule");
+      final MutableProtocolSchedule protocolSchedule =
+          (MutableProtocolSchedule) spec.getConfigOptions();
```

Comment:
Is getConfigOptions() an appropriate method name now that BftForkSpec<C> is more generic?

Reply:
No probably not, I had the same thought. But didn't want to do any refactoring of that class in this PR as I was expecting a lot of other files to change from a rename. But I'll try it out locally it might not be that bad.

Reply:
There was an additional 10 files added to PR. So will do that refactoring in another PR